rmmxapianapi.cc

```diff
@@ -85,13 +85,34 @@ class DatabaseContainer {
       }
       return writabledatabase;
     }
+
+    Xapian::Document getDocumentByUniqueTerm(char * unique_term, Xapian::WritableDatabase * writabledatabasePtr) {
+      Xapian::Document doc;   
```

Comment:
Won't this return a potentially undefined value if the term is neither in dbh nor in the addedWritableDatabases? Can that be a problem?

Reply:
Thanks for pointing out. I've made two test cases to demonstrate what happens if trying to modify documents by ids that does not exist in the database:

https://github.com/runbox/runbox-searchindex/pull/1/files#diff-96201e983bb2d14d1d4e808151d45874R132

As you can see an exception is thrown to JS, and for the second test case I actually add a document with the non-existing ID after first trying to modify it - and it still plays.

One could avoid exceptions being thrown by adding an error handler (`if(doc==NULL)`), but still JS would have to be notified that things went wrong, so that would just be more code for obtaining the same. Relying on exceptions being thrown means less code and more readable code (I agree to what's written here: https://isocpp.org/wiki/faq/exceptions#why-exceptions)

Regarding the exceptions thrown, it's only an error number when using optimized emscripten builds ( https://emscripten.org/docs/optimizing/Optimizing-Code.html#other-optimization-issues ), but it should be possible to get more info in a debugging scenario if turning off optimizations.

But thanks for pointing out, and would be happy to get your comment if you think that I've missed something in my reasoning here.